sdks/python/apache_beam/examples/complete/check_requirements_file_packages.py

```diff
@@ -0,0 +1,71 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A pipeline to verify the installation of packages specified in the
+   requirements.txt.
+   python check_requirements_file_packages.py \\
+      --runner=PortableRunner \\
+      --job_endpoint=embed \\
+      --requirements_file=./requirements.txt \\
+      --environment_type="DOCKER"
+"""
+
+import argparse
+import apache_beam as beam
+import logging
+
+from apache_beam.options.pipeline_options import PipelineOptions
+
+
+def run(argv=None):
+
+  _PACKAGE_IN_REQUIREMENTS_FILE = 'matplotlib'
```

Comment:
you many need to pass --save_main_session for these constants to be available to the runner, since they are defined outside DoFns that use them.
FWIW this could be simplified to smth like

```
def verify_packages_from_requirements_file_are_installed(unused_element):
	... verify matplotlib is installed ...

beam.Create([None]) | beam.Map(verify_packages_from_requirements_file_are_installed)
```

Reply:
I tried that first but then it was throwing errors like <variable> not defined. So I had to change the code. May be `--saving_main_session` could solve the issue

Reply:
not sure what errors you saw. in this approach everything can be defined within `verify_packages_from_requirements_file_are_installed` so there should be no issues.

Reply:
I put the global variables out of the method scope, now `--save_main_session` worked. Thanks